packages/core/src/loaders/imageLoader.ts

```diff
@@ -183,6 +197,122 @@ export function loadAndCacheImages(
   return allPromises;
 }
 
+/**
+ * Loads an image given an imageId and optional priority and returns a promise
+ * which will resolve to the loaded image object or fail if an error occurred.
+ * The image is stored in the cache.
+ *
+ * @param imageId -  A Cornerstone Image Object's imageId
+ * @param options - Options to be passed to the Image Loader
+ *
+ * @returns Image Loader Object
+ */
+export function createAndCacheDerivedImage(
+  imageId: string,
+  options: DerivedImageOptions
+): Promise<IImage> {
+  if (imageId === undefined) {
+    throw new Error(
+      'createAndCacheDerivedImage: parameter imageId must not be undefined'
+    );
+  }
+  const imagePlaneModule = metaData.get('imagePlaneModule', imageId);
+
+  let imageScalarData;
+  const length = imagePlaneModule.rows * imagePlaneModule.columns;
+  let numBytes, TypedArray;
+
+  const { targetBuffer } = options;
+  if (targetBuffer) {
+    if (targetBuffer.type === 'Float32Array') {
+      numBytes = length * 4;
+      TypedArray = Float32Array;
+    } else if (targetBuffer.type === 'Uint8Array') {
+      numBytes = length;
+      TypedArray = Uint8Array;
+    } else if (targetBuffer.type === 'Uint16Array') {
+      numBytes = length * 2;
+      TypedArray = Uint16Array;
+    } else if (targetBuffer.type === 'Int16Array') {
+      numBytes = length * 2;
+      TypedArray = Uint16Array;
+    } else {
+      throw new Error('TargetBuffer should be Float32Array or Uint8Array');
+    }
+  } else {
+    // Use Uint8 if no targetBuffer is provided
+    numBytes = length;
+    TypedArray = Uint8Array;
+  }
+
+  if (targetBuffer?.sharedArrayBuffer) {
+    const buffer = new SharedArrayBuffer(numBytes);
```

Comment:
we don't need sharedArrayBuffer for images

Reply:
I was thinking about very large images (like tomosynthesis). Anyway its not a default option

Reply:
SAB doesn't have any benefit regarding size